tests/Query/BuilderTest.php

```diff
@@ -1599,29 +1599,86 @@ public static function getEloquentMethodsNotSupported()
         yield 'orWhereIntegerNotInRaw' => [fn (Builder $builder) => $builder->orWhereIntegerNotInRaw('id', ['1a', 2])];
     }
 
-    public function testRenameEmbeddedIdFieldCanBeDisabled()
+    #[DataProvider('provideDisableRenameEmbeddedIdField')]
+    public function testDisableRenameEmbeddedIdField(array $expected, Closure $build)
     {
         $builder = $this->getBuilder(false);
         $this->assertFalse($builder->getConnection()->getRenameEmbeddedIdField());
 
-        $mql = $builder
-            ->where('id', '=', 10)
-            ->where('nested.id', '=', 20)
-            ->where('embed', '=', ['id' => 30])
-            ->toMql();
-
-        $this->assertEquals([
-            'find' => [
-                [
-                    '$and' => [
-                        ['_id' => 10],
-                        ['nested.id' => 20],
-                        ['embed' => ['id' => 30]],
+        $mql = $build($builder)->toMql();
+
+        $this->assertEquals($expected, $mql);
+    }
+
+    public static function provideDisableRenameEmbeddedIdField()
+    {
+        yield 'rename embedded id field' => [
+            [
+                'find' => [
+                    [
+                        '$and' => [
+                            ['_id' => 10],
+                            ['nested.id' => 20],
+                            ['embed' => ['id' => 30]],
+                        ],
+                    ],
+                    ['typeMap' => ['root' => 'object', 'document' => 'array']],
+                ],
+            ],
+            fn (Builder $builder) => $builder->where('id', '=', 10)
+                ->where('nested.id', '=', 20)
+                ->where('embed', '=', ['id' => 30]),
+        ];
+
+        yield 'rename root id' => [
+            ['find' => [['_id' => 10], ['typeMap' => ['root' => 'object', 'document' => 'array']]]],
+            fn (Builder $builder) => $builder->where('id', '=', 10),
+        ];
+
+        yield 'nested id not renamed' => [
+            ['find' => [['nested.id' => 20], ['typeMap' => ['root' => 'object', 'document' => 'array']]]],
+            fn (Builder $builder) => $builder->where('nested.id', '=', 20),
+        ];
+
+        yield 'embed id not renamed' => [
+            ['find' => [['embed' => ['id' => 30]], ['typeMap' => ['root' => 'object', 'document' => 'array']]]],
+            fn (Builder $builder) => $builder->where('embed', '=', ['id' => 30]),
+        ];
+
+        yield 'nested $and in $or' => [
+            [
+                'find' => [
+                    [
+                        '$or' => [
+                            [
+                                '$and' => [
+                                    ['_id' => 10],
```

Comment:
There are flaws in the conversion if you use advanced queries. This transformation from `id` to `_id` is done in the `compileWheres` from the “column name”: https://github.com/mongodb/laravel-mongodb/blob/bdae18d817483d14d200735f00b3d9d43d75e3f4/src/Query/Builder.php#L1209-L1212